src/pure/List.mo

```diff
@@ -61,8 +60,27 @@ module {
       case (?(_, t)) 1 + size t
     };
 
-  public func contains<T>(list : List<T>, item : T) : Bool {
-    todo()
+  /// Check if the list contains a given value. Uses the provided equality function to compare values.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// import Nat "mo:base/Nat";
+  /// List.contains<Nat>(?(1, ?(2, ?(3, null))), Nat.equal, 2) // => true
+  /// ```
+  ///
+  /// Runtime: O(size)
+  ///
+  /// Space: O(1)
+  ///
+  /// *Runtime and space assumes that `equal` runs in O(1) time and space.
+  public func contains<T>(list : List<T>, equal : (T, T) -> Bool, item : T) : Bool {
```

Comment:
Interestingly `pure/List.mo` has no `equals()` while `pure/Queue.mo` does have one. Does the API validator miss out on this?

Reply:
Also, the signature of `pure/Queue.contains` needs a fixup.

Reply:
At one point I think there was a discussion about removing the `equal()` function for some modules, but I think it makes sense to include this. Updated the PR to require `equal()` for data structure modules as well as adjusting `pure/Queue`. Feel free to change the API as needed.